recipe/reader.go

```diff
@@ -4,28 +4,45 @@ import (
 	"bytes"
 	"os"
 	"path/filepath"
-	"strings"
 	"text/template"
 
 	"gopkg.in/yaml.v3"
 )
 
-var (
-	recipeEnvVarPrefix = "METEOR_"
-)
-
 type Reader struct {
 	data map[string]string
 }
 
 func NewReader() *Reader {
 	reader := &Reader{}
-	reader.populateData()
+	reader.data = populateData()
 
 	return reader
 }
 
-func (r *Reader) Read(path string) (recipe Recipe, err error) {
+//  Read loads the list of recipes from a give file or directory path.
+func (r *Reader) Read(path string) (recipes []Recipe, err error) {
+	fi, err := os.Stat(path)
```

Comment:
Small suggestion, although current implementation is great we can also think of using a file system interface abstraction like [afero](https://github.com/spf13/afero). This will also allow in-memory filesystem really helpful in testing.